tests/queries/0_stateless/02861_alter_replace_partition_do_not_wait_mutations_on_unrelated_partitions.sql

```diff
@@ -0,0 +1,36 @@
+-- https://github.com/ClickHouse/ClickHouse/issues/45328
+-- Check that replacing one partition on a table with `ALTER TABLE REPLACE PARTITION`
+-- doesn't wait for mutations on other partitions.
+-- { echo }
+
+DROP TABLE IF EXISTS t1;
+DROP TABLE IF EXISTS t2;
+
+CREATE TABLE t1
+(
+    `p` UInt8,
+    `i` UInt64
+)
+ENGINE = MergeTree
+PARTITION BY p
+ORDER BY tuple();
+
+INSERT INTO t1 VALUES (1, 1), (2, 2);
+
+CREATE TABLE t2 AS t1;
+INSERT INTO t2 VALUES (2, 2000);
+
+-- mutation that is supposed to be running in bg while REPLACE is performed.
+-- sleep(1) is arbitrary here, we just need mutation to take long enough to be noticeable.
+ALTER TABLE t1 UPDATE i = if(sleep(1), 0, 9000) IN PARTITION id '1' WHERE p == 1;
+-- check that mutation is started
+SELECT not(is_done) as is_running FROM system.mutations WHERE database==currentDatabase() AND table=='t1';
```

Comment:
It is possible that this script will freeze for more than one second between these two statements, and then the test will fail.

Reply:
Hmm, this needs a good thought...

Reply:
.sh test with repeats will be ok.

It can be made in a way so if the change is present, it will succeed in a certain number of tries, and if the change is not present it is guaranteed to not succeed.